packages/surveys/src/components/MultipleChoiceMultiQuestion.tsx

```diff
@@ -77,7 +86,11 @@ export default function MultipleChoiceSingleQuestion({
     <form
       onSubmit={(e) => {
         e.preventDefault();
-        onSubmit({ [question.id]: value });
+        const newValue = (value as string[]).filter((item) => {
+          return getChoicesWithoutOtherLabels().includes(item) || item === otherValue;
+        }); // filter out all those values which are either in getChoicesWithoutOtherLabels() (i.e. selected by checkbox) or the latest entered otherValue
+        onChange({ [question.id]: newValue });
```

Comment:
and do we need this here since we already have an onSubmit?

Please test and see if its needed or not

Reply:
Sure, will check and update.

Reply:
onChange is required for local updates.